src/document/docUtil.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * Finds the closest ancestor DOM element with the specified class.
+ */
+export function findClosestElWithClass(element: Element | null, className: string): Element | null {
+    for (let el = element; el && el !== document.body; el = el.parentNode as Element | null) {
+        if (el.classList && el.classList.contains(className)) {
+            return el;
+        }
+    }
+
+    return null;
+}
+
+/**
+ * Returns the page element and page number that the element is on.
+ */
+export function getPageNumber(element: Element | null): number | undefined {
+    const pageEl = findClosestElWithClass(element, 'page');
+    const pageNumber = pageEl && pageEl.getAttribute('data-page-number');
+
+    return pageNumber ? parseInt(pageNumber, 10) : undefined;
+}
+
+export function getRange(): Range | null {
+    const selection = window.getSelection();
+    if (!selection || selection.type !== 'Range' || selection.isCollapsed) {
+        return null;
+    }
+
+    return selection.getRangeAt(0);
+}
+
+export function getSelection(): {
```

Comment:
The return type is a little difficult to read. You could move it to an alias that lives at the top of this module.